### DISCUSSION TITLE
What's the best way to pass props from a parent component to a child route?

### DISCUSSION THREAD

**Discussion by mikemorton** [TanStack/router#471] [Q&A]
I'm guessing I could do this with a context provider and `useContext` but I'm wondering if there is a better way built into the router already?

**Comment by sachingaikwad123** [comment]
I am also looking for answer for this. For example if I am using <Link> then I can pass "param" (route param) - which one one parameter. Child route component may expect more props. How to pass them from <Link>?

I found following ways:

- Pass props via "state" and then access them in child component from useLocation()

Is there any other better way than this?
